### PR TITLE
General improvements for the front-end

### PR DESCRIPTION
**Pull request by Senyoret1**
This PR contains changes that were pending in the SkycoinProject repo. The changes are old, so some modiffications were added to make them work with the current state of the repo. The changes are:

-	The uncommented code files are now commented.
-	Some problems with the popup for editting the wallet label were solved.
-	Improvements for the code that blocks the form for creating wallets while checks are being performed.
-	Improvements for the modal window used for selecting how many addresses to add to a wallet.
-	Code improvements for the wallet details component and changes for preventing it to enter in an invalid state.
-	Code for cleaning events was added.
-	A problem with the mouse pointer of the wallet option buttons was solved.
-	Small improvements for the feedback shown when clicking buttons.
-	The msg shown when an operation with the Skywallet is cancelled was improved.
-	The routing and translation configs were moved to idependent modules.
-	The amount pipe was improved to process BigNumber instances directly.
-	Improvements for managing the CORS policy when connecting to Swablab, from Electron and the Angular test server, were added.
-	Improvements and bug fixes for the seed management in the wallet creation form.
-	Several other smaller fixes and improvements.

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

File: src/gui/static/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      },
+    ],
+    canActivate: [WizardGuardService],
+  },
+  {
+    path: 'wizard',
+    component: OnboardingComponent,
+  },
+  {
+    path: 'reset/:id',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:85]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
    path: "reset/:id",
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import { WalletsComponent } from './components/pages/wallets/wallets.component';
+import { SendSkycoinComponent } from './components/pages/send-skycoin/send-skycoin.component';
+import { RouterModule } from '@angular/router';
+import { PendingTransactionsComponent } from './components/pages/settings/pending-transactions/pending-transactions.component';
+import { OutputsComponent } from './components/pages/settings/outputs/outputs.component';
+import { BlockchainComponent } from './components/pages/settings/blockchain/blockchain.component';
+import { BackupComponent } from './components/pages/settings/backup/backup.component';
+import { NetworkComponent } from './components/pages/settings/network/network.component';
+import { BuyComponent } from './components/pages/buy/buy.component';
+import { TransactionListComponent } from './components/pages/transaction-list/transaction-list.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { TransactionListComponent } from "./components/pages/transaction-list/transaction-list.component";
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import { NgModule } from '@angular/core';
+import { WalletsComponent } from './components/pages/wallets/wallets.component';
+import { SendSkycoinComponent } from './components/pages/send-skycoin/send-skycoin.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { SendSkycoinComponent } from "./components/pages/send-skycoin/send-skycoin.component";
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+
+const ROUTES = [
+  {
+    path: '',
+    redirectTo: 'wallets',
+    pathMatch: 'full',
+  },
+  {
+    path: 'wallets',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:25]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
    path: "wallets",
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import { PendingTransactionsComponent } from './components/pages/settings/pending-transactions/pending-transactions.component';
+import { OutputsComponent } from './components/pages/settings/outputs/outputs.component';
+import { BlockchainComponent } from './components/pages/settings/blockchain/blockchain.component';
+import { BackupComponent } from './components/pages/settings/backup/backup.component';
+import { NetworkComponent } from './components/pages/settings/network/network.component';
+import { BuyComponent } from './components/pages/buy/buy.component';
+import { TransactionListComponent } from './components/pages/transaction-list/transaction-list.component';
+import { WizardGuardService } from './services/wizard-guard.service';
+import { OnboardingComponent } from './components/pages/onboarding/onboarding.component';
+import { ResetPasswordComponent } from './components/pages/reset-password/reset-password.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:14]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { ResetPasswordComponent } from "./components/pages/reset-password/reset-password.component";
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import { NgModule } from '@angular/core';
+import { WalletsComponent } from './components/pages/wallets/wallets.component';
+import { SendSkycoinComponent } from './components/pages/send-skycoin/send-skycoin.component';
+import { RouterModule } from '@angular/router';
+import { PendingTransactionsComponent } from './components/pages/settings/pending-transactions/pending-transactions.component';
+import { OutputsComponent } from './components/pages/settings/outputs/outputs.component';
+import { BlockchainComponent } from './components/pages/settings/blockchain/blockchain.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { BlockchainComponent } from "./components/pages/settings/blockchain/blockchain.component";
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      },
+      {
+        path: 'pending-transactions',
+        component: PendingTransactionsComponent,
+      },
+    ],
+    canActivate: [WizardGuardService],
+  },
+  {
+    path: 'wizard',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:81]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
    path: "wizard",
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+import { RouterModule } from '@angular/router';
+import { PendingTransactionsComponent } from './components/pages/settings/pending-transactions/pending-transactions.component';
+import { OutputsComponent } from './components/pages/settings/outputs/outputs.component';
+import { BlockchainComponent } from './components/pages/settings/blockchain/blockchain.component';
+import { BackupComponent } from './components/pages/settings/backup/backup.component';
+import { NetworkComponent } from './components/pages/settings/network/network.component';
+import { BuyComponent } from './components/pages/buy/buy.component';
+import { TransactionListComponent } from './components/pages/transaction-list/transaction-list.component';
+import { WizardGuardService } from './services/wizard-guard.service';
+import { OnboardingComponent } from './components/pages/onboarding/onboarding.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:13]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { OnboardingComponent } from "./components/pages/onboarding/onboarding.component";
```

File: src/gui/static/src/app/components/pages/exchange/exchange-status/exchange-status.component.ts [hunk 9]
```diff
@@ -1,22 +1,32 @@
 import { Component, Input, OnDestroy, Output, EventEmitter } from '@angular/core';
-import { ExchangeOrder, StoredExchangeOrder } from '../../../../app.datatypes';
-import { ExchangeService } from '../../../../services/exchange.service';
 import { MatDialog } from '@angular/material/dialog';
 import { SubscriptionLike, of } from 'rxjs';
+import { delay, mergeMap } from 'rxjs/operators';
+
+import { ExchangeService, ExchangeOrder, StoredExchangeOrder } from '../../../../services/exchange.service';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/exchange/exchange-status/exchange-status.component.ts:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { ExchangeService, ExchangeOrder, StoredExchangeOrder } from "../../../../services/exchange.service";
```

File: src/gui/static/src/app/components/pages/exchange/exchange-create/exchange-create.component.ts [hunk 10]
```diff
@@ -45,25 +48,30 @@ export class ExchangeCreateComponent implements OnInit, OnDestroy {
...
   private exchangeSubscription: SubscriptionLike;
   private priceUpdateSubscription: SubscriptionLike;
 
-  get toAmount() {
+  // Approximately how many coins will be received for the amount of coins the user will send,
+  // as per the value entered on the form and the current price.
+  get toAmount(): string {
     if (!this.activeTradingPair) {
-      return 0;
+      return '0';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/exchange/exchange-create/exchange-create.component.ts:62]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
      return "0";
```

File: src/gui/static/src/app/components/pages/wallets/wallet-detail/wallet-detail.component.ts [hunk 11]
```diff
@@ -1,26 +1,30 @@
...
+import { MatDialog } from '@angular/material/dialog';
 import { TranslateService } from '@ngx-translate/core';
-import { HwWalletService } from '../../../../services/hw-wallet.service';
 import { Observable, SubscriptionLike } from 'rxjs';
+import { Router } from '@angular/router';
+import { mergeMap, first } from 'rxjs/operators';
+
+import { ChangeNameComponent, ChangeNameData } from '../change-name/change-name.component';
+import { PasswordDialogComponent, PasswordDialogParams, PasswordSubmitEvent } from '../../../layout/password-dialog/password-dialog.component';
+import { NumberOfAddressesComponent, NumberOfAddressesEventData } from '../number-of-addresses/number-of-addresses';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/wallets/wallet-detail/wallet-detail.component.ts:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { NumberOfAddressesComponent, NumberOfAddressesEventData } from "../number-of-addresses/number-of-addresses";
```

File: src/gui/static/src/app/services/exchange.service.ts [hunk 12]
```diff
@@ -1,49 +1,158 @@
...
+import { map, mergeMap, retryWhen, delay, catchError, tap } from 'rxjs/operators';
 import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders, HttpErrorResponse } from '@angular/common/http';
-import {
-  ExchangeOrder,
-  StoredExchangeOrder,
-  TradingPair,
-} from '../app.datatypes';
-import { StorageService, StorageType } from './storage.service';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/services/exchange.service.ts:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { HttpClient, HttpHeaders } from "@angular/common/http";
```

File: src/gui/static/src/app/components/pages/wallets/change-name/change-name.component.ts [hunk 13]
```diff
@@ -13,42 +13,76 @@ import { SoftwareWalletService } from '../../../../services/wallet-operations/so
...
-  Initial,
-  WaitingForConfirmation,
+  /**
+   * Showing the form.
+   */
+  Initial = 'Initial',
+  /**
+   * Asking the user to confirm the operation on the hw wallet.
+   */
+  WaitingForConfirmation = 'WaitingForConfirmation',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/wallets/change-name/change-name.component.ts:27]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
  WaitingForConfirmation = "WaitingForConfirmation",
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 14]
```diff
@@ -0,0 +1,89 @@
+import { NgModule } from '@angular/core';
+import { WalletsComponent } from './components/pages/wallets/wallets.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { WalletsComponent } from "./components/pages/wallets/wallets.component";
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+import { WizardGuardService } from './services/wizard-guard.service';
+import { OnboardingComponent } from './components/pages/onboarding/onboarding.component';
+import { ResetPasswordComponent } from './components/pages/reset-password/reset-password.component';
+import { ExchangeComponent } from './components/pages/exchange/exchange.component';
+
+
+const ROUTES = [
+  {
+    path: '',
+    redirectTo: 'wallets',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:21]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
    redirectTo: "wallets",
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 16]
```diff
@@ -0,0 +1,89 @@
+import { NgModule } from '@angular/core';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { NgModule } from "@angular/core";
```

File: src/gui/static/src/app/components/pages/wallets/change-name/change-name.component.ts [hunk 17]
```diff
@@ -13,42 +13,76 @@ import { SoftwareWalletService } from '../../../../services/wallet-operations/so
...
+/**
+ * States in which ChangeNameComponent can be.
+ */
 enum States {
-  Initial,
-  WaitingForConfirmation,
+  /**
+   * Showing the form.
+   */
+  Initial = 'Initial',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/wallets/change-name/change-name.component.ts:23]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
  Initial = "Initial",
```

File: src/gui/static/src/app/components/pages/exchange/exchange.component.ts [hunk 18]
```diff
@@ -1,21 +1,28 @@
 import { Component, OnInit, OnDestroy } from '@angular/core';
-import { StoredExchangeOrder } from '../../../app.datatypes';
-import { ExchangeService } from '../../../services/exchange.service';
 import { MatDialog } from '@angular/material/dialog';
-import { ExchangeHistoryComponent } from './exchange-history/exchange-history.component';
 import { SubscriptionLike } from 'rxjs';
+
+import { ExchangeService, StoredExchangeOrder } from '../../../services/exchange.service';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/exchange/exchange.component.ts:5]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { ExchangeService, StoredExchangeOrder } from "../../../services/exchange.service";
```

File: src/gui/static/src/app/services/exchange.service.ts [hunk 19]
```diff
@@ -1,49 +1,158 @@
...
 import * as moment from 'moment';
-import { ApiService } from './api.service';
+
+import { StorageService, StorageType } from './storage.service';
 import { environment } from '../../environments/environment';
-import { map, mergeMap, retryWhen, delay, catchError, tap } from 'rxjs/operators';
 import { processServiceError } from '../utils/errors';
+import { OperationError } from '../utils/operation-error';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/services/exchange.service.ts:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { OperationError } from "../utils/operation-error";
```

File: src/gui/static/src/app/app.config.ts [hunk 20]
```diff
@@ -1,15 +1,74 @@
...
+  /**
+   * ID of the coin on the coin price service. If null, the wallet will not show the USD price.
+   */
+  priceApiId: 'sky-skycoin',
+  /**
+   * This wallet uses the Skycoin URI Specification (based on BIP-21) when creating QR codes and
+   * requesting coins. This variable defines the prefix that will be used for creating QR codes
+   * and URLs. IT MUST BE UNIQUE FOR EACH COIN.
+   */
+  uriSpecificatioPrefix: 'skycoin',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app.config.ts:30]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
  uriSpecificatioPrefix: "skycoin",
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+import { WalletsComponent } from './components/pages/wallets/wallets.component';
+import { SendSkycoinComponent } from './components/pages/send-skycoin/send-skycoin.component';
+import { RouterModule } from '@angular/router';
+import { PendingTransactionsComponent } from './components/pages/settings/pending-transactions/pending-transactions.component';
+import { OutputsComponent } from './components/pages/settings/outputs/outputs.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { OutputsComponent } from "./components/pages/settings/outputs/outputs.component";
```

File: src/gui/static/src/app/app-routing.module.ts [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+import { BlockchainComponent } from './components/pages/settings/blockchain/blockchain.component';
+import { BackupComponent } from './components/pages/settings/backup/backup.component';
+import { NetworkComponent } from './components/pages/settings/network/network.component';
+import { BuyComponent } from './components/pages/buy/buy.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/app-routing.module.ts:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { BuyComponent } from "./components/pages/buy/buy.component";
```

**Comment by Senyoret1** [comment]
@SkycoinSynth The linter rules of the front-end are different from the Codacy rules. Should the project be adapted to the Codacy rules?

**Comment by iketheadore** [comment]
@Senyoret1 please resolve conflicts, thanks.

**Comment by iketheadore** [comment]
Looks good to me 👍

**Pull request merged**